lib/auth/new_web_user.go

```diff
@@ -40,6 +40,19 @@ import (
 	"github.com/tstranex/u2f"
 )
 
+func (s *AuthServer) getTOTPLocation() string {
+	clusterName, err := s.GetClusterName()
+	if err == nil {
+		return clusterName.GetClusterName()
+	}
+	proxies, err := s.GetProxies()
+	if err == nil {
+		return proxies[0].GetPublicAddr()
```

Comment:
This will fail if you get an empty list of proxies.

@kimlisa Is working on a function that will fetch the address of the proxy in https://github.com/gravitational/teleport/pull/3476. Once that's merged, can you call that function?

If you don't want to do that, leave a comment here and I'll do it for you and merge this PR. I'll leave your original two commits and just apply a commit on top of that.

Reply:
@russjones the function you are referring to is in `hornet` branch. That not going to be merged any time soon. @kimlisa will create a ticket to be addressed once `hornet` gets merged to master.

Reply:
pinging @webvictim @benarent to make sure this is what they wanted:

Gus and Ben said on the issue:

> I think it would make much more sense if the default 'caption' or name offered when scanning the QR code matches the value of proxy_service.public_addr if set

They said `if set`. And to me `if set` means if the admin specified it on the config. From my experience so far, if `proxy_service.public_addr` is NOT set, `GetProxies` will **still** return a list of proxies, just `proxies[0].GetPublicAddr()` will be an **empty value**.

The function that @russjones wanted me to make, will check for this empty value, and will replace empty value with a manually built address using default values (which is hostname:defaultPort). And I don't think this is what they wanted?

I think what they wanted was this:
```
proxies, err := s.GetProxies()
if len(proxies) > 0 && proxies[0].GetPublicAddr() != "" {
        // definitely set by user
	return proxies[0].GetPublicAddr();
}
```

Ben/Gus also says `if set` for cluster name. In the guide it says if the `cluster_name` is NOT set: `By default an automatically generated name is used (not recommended)`. Btw, I don't think this is correct statement b/c I see it set to my hostname:

```
// init cfg
// if user did not provide auth domain name, use this host's name
if cfg.Auth.Enabled && cfg.Auth.ClusterName == nil {
	cfg.Auth.ClusterName, err = services.NewClusterName(services.ClusterNameSpecV2{
		ClusterName: cfg.Hostname,
	})
	if err != nil {
		return nil, trace.Wrap(err)
	}
}
```

I'm not too sure `GetClusterName()` checks that it was set. As seen above in the init phase, it looks like it gets replaced with a default value,  which is hostname. So `GetClusterName` will always return a hostname, if not explicitly set by user:

```
clusterName, err := s.GetClusterName()
if err == nil {
	return clusterName.GetClusterName()
}
```
This means checking for `GetClusterName` should be checked last.

Sorry that was long. But I wanted to make sure I understood the problem being solved.

Reply:
Hey @kimlisa, I'm just following up on this. 

I think we should.
1. Say that we'll set cluster_name to hostname if that's the default behavior. 
2. Improve the display name of cluster. https://github.com/gravitational/teleport/issues/3604